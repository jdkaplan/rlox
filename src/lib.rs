macro_rules! debug_log_gc {
    ($($arg:tt)*) => {{
        #[cfg(feature = "log_gc")]
        println!($($arg)*);
    }};
}

mod chunk;
mod compiler;
mod gc;
mod heap;
mod object;
mod scanner;
mod table;
mod value;
mod vm;

pub use vm::{InterpretError, InterpretResult, Vm};

pub const U8_COUNT: usize = (u8::MAX as usize) + 1;
pub const FRAMES_MAX: usize = 64;
pub const STACK_MAX: usize = FRAMES_MAX * U8_COUNT;
