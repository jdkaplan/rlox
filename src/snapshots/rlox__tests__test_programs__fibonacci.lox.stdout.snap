---
source: src/lib.rs
description: "fun fibonacci(n) {\n    if (n <= 1) return 1;\n    return fibonacci(n - 1) + fibonacci(n - 2);\n}\n\nfor (var i = 0; i < 10; i = i + 1) {\n    print fibonacci(i);\n}\n"
input_file: test_programs/fibonacci.lox
---
1
1
2
3
5
8
13
21
34
55

